File: java/client/src/main/java/org/apache/rocketmq/client/java/example/LiteProducerExample.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package org.apache.rocketmq.client.java.example;

import java.nio.charset.StandardCharsets;
import org.apache.rocketmq.client.apis.ClientException;
import org.apache.rocketmq.client.apis.ClientServiceProvider;
import org.apache.rocketmq.client.apis.message.Message;
import org.apache.rocketmq.client.apis.producer.Producer;
import org.apache.rocketmq.client.apis.producer.SendReceipt;
import org.apache.rocketmq.client.java.exception.LiteTopicQuotaExceededException;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class LiteProducerExample {
    static final Logger log = LoggerFactory.getLogger(LiteProducerExample.class);

    private LiteProducerExample() {
    }

    public static void main(String[] args) throws ClientException {
        final ClientServiceProvider provider = ClientServiceProvider.loadService();

        String topic = "yourParentTopic";
        final Producer producer = ProducerSingleton.getInstance(topic);
        // Define your message body.
        byte[] body = "This is a lite message for Apache RocketMQ".getBytes(StandardCharsets.UTF_8);
        final Message message = provider.newMessageBuilder()
            // Set topic for the current message.
            .setTopic(topic)
            // Key(s) of the message, another way to mark message besides message id.
            .setKeys("yourMessageKey-3ee439f945d7")
            // Set your lite topic
            .setLiteTopic("lite-topic-1")
            .setBody(body)
            .build();
        try {
            final SendReceipt sendReceipt = producer.send(message);
            log.info("Send message successfully, messageId={}", sendReceipt.getMessageId());
        } catch (LiteTopicQuotaExceededException e) {
            // Lite topic quota exceeded.
            // Evaluate and increase the lite topic resource limit.
            log.error("Lite topic quota exceeded", e);
        } catch (Throwable t) {
            log.error("Failed to send message", t);
        }
        // Close the producer when you don't need it anymore.
        // You could close it manually or add this into the JVM shutdown hook.
        // producer.close();
    }
}

File: php/examples/LiteProducerExample.php
<?php
/**
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

require_once __DIR__ . '/../vendor/autoload.php';
require_once __DIR__ . '/../Producer.php';
require_once __DIR__ . '/../Logger.php';
require_once __DIR__ . '/ExampleConfig.php';

use Apache\Rocketmq\Producer;
use Apache\Rocketmq\SessionCredentials;
use Apache\Rocketmq\V2\Message;
use Apache\Rocketmq\V2\Resource;
use Apache\Rocketmq\V2\SystemProperties;

$config = ExampleConfig::getInstance();
$endpoints = $config->getEndpoints();
$topic = $config->getTopic('priority');
$credentials = $config->getCredentials();

$producer = new Producer($endpoints, [
    'topics' => [$topic],
    'credentials' => $credentials,
    'maxAttempts' => 3,
    'requestTimeout' => 3000,
]);

$producer->start();

$topicResource = new Resource();
$topicResource->setName($topic);

$sysProps = new SystemProperties();
$sysProps->setKeys(['yourMessageKey-3ee439f945d7']);
// Set your lite topic as a sub-classifier under the parent topic
$sysProps->setLiteTopic('lite-topic-1');

$message = new Message();
$message->setTopic($topicResource);
$message->setBody('This is a lite message for Apache RocketMQ');
$message->setSystemProperties($sysProps);

try {
    $result = $producer->send($message);
    echo "Send message successfully, messageId=" . $result['messageId'] . "\n";
} catch (\Throwable $e) {
    echo "Failed to send message: " . $e->getMessage() . "\n";
}

$producer->shutdown();
